### ISSUE TITLE
Global refactoring and project board organization

### ISSUE THREAD

**Issue by FolkertVanVerseveld** [FolkertVanVerseveld/aoe#18]
Development has been suspended for almost 6 months on the public repo
because the reverse enginering process took much longer than expected. I
am also very busy irl but I am doing my best to resume this project step
by step.

I have been thinking for a long time about the best way to continue this
project. There is a lot of ugly code and even some broken stuff as well.
I am currently thinking about the following:

- [x] Properly document the current code and improve the maintainability.
- [x] Remove outdated components.

And also the following much more time consuming points:

- [x] Redesign the build process.
- [x] Create a project board to keep track of the overal progress and the
    direction of the project.
- [x] Document file formats from the original genie engine to make sure
    other projects can benefit as well.
- [x] Port old game and genie engine to empires/
  - [x] Redesign User Interface in C++.
  - [x] Use libpe for PE library resource fetching.

Even though multiple other projects have similar goals to this project,
I have noticed a long time ago that each project has their own problems.

The last thing I am writing down to make sure I don't forget it: make
sure that even slow old computers, maybe even as old as computers with
the original soft- and hardware requirements, are able to run this game.

**Comment by FolkertVanVerseveld** [comment]
About 3 more months have passed without much change, but finally I've
removed a lot of unused obsolete code and added a simple proof of
concept port of aoesetup.exe. It is currently located in setup/ and
lacks documentation, but it should not too much work to build it if you
want to.

I may or may not push force in the near future to get rid of a couple of
old commits even though I don't like messing around with old commits.

The maintainability is one of the most important points right now,
because this has prevented me in the last 10 months or so from seriously
continuing the project.

**Comment by FolkertVanVerseveld** [comment]
A new game launcher has been added as well as a global refactoring and
porting phase to upgrade the old game and genie game engine. The
empires/ folder already is much easier to configure and maintain.

The new game rendering engine is very different from the old one. For
one it is using the old stock SDL_Renderer rather than OpenGL which is
not a very good thing... but a OpenGL rendering backend will be added
once more game mechanics have been added.

I've also finally decided to write the User Interface and game mechanics
primarly in C++, because this greatly increases productivity and the
original game also used C++ for this... At the moment I'm sticking with
C++11 because nowadays, most compilers are compliant enough with the
standard and we don't need much new C++ stuff anyway.

It is still going to take a while to be able to get the code quality to
something more acceptable, but bear with me. I can do this! I just need
some more time...

PS: I've updated the first post to reflect these changes
